### PR TITLE
ggiraph demo

### PR DESCRIPTION
**Pull request by rmbielby**
## Pull request overview

Testing out ggiraph as an alternative to plotly

## Pull request checklist

Please check if your PR fulfils the following:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Tests have been run locally and are passing (`run_tests_locally()`)
- [x] Code is styled according to tidyverse styling (checked locally with `tidy_code()`)

## What is the current behaviour?

We've got two charts, both of which are rendered using plotly to allow interactivity (and ggplot to produce the underlying charts). I've always found plotly to be a bit awkward in terms of setting the chart sizing and feeding through tooltips properly.
 
## What is the new behaviour?

I've switched both charts to be rendered using ggiraph and played around with tooltips and setting the plot size. Generally feels like an easier experience. I found it useful to set the width and height as 100% in girafeOutput() and then set the aspect-ratio / scaling in renderGirafe() using the width_svg and height_svg keywords. Don't think there's a way to dynamically set the aspect ration, but I find the above works well for filling the available space. It'll default to filling the available width provided by the parent container as I've got it now.

I added some example tooltips as well, which was really straightforward.

## Anything else

I've done some clean-up around the layout whilst updating the plots:
 - took away the use of box(), which was pretty redundant and just added unnecessary filler both in the code and in the resulting UI
 - Split the first tab into 2 separate tabs, so that the user doesn't have to scroll to find the chart now.
 - Cleaned out renv a little. Think there were a few extraneous packages (as well as plotly becoming redundant)
 - Used the str_wrap() function from stringr to display the x-axis labels a little more neatly in the bar chart
 - Whilst updating the charts, I figured it would be useful to us (and people using the template) if we had a couple more UI tests on it in addition to the load test, so have added those on.

### CODE REVIEW

**Comment by chfoster** [comment]
Can we add the values of the line to the hover-over box on this one? 
image

File: R/plotting.R [hunk 1]
```diff
@@ -5,22 +5,24 @@ createAvgRevTimeSeries <- function(df, inputArea) {
...
+      axis.title.y = element_text(angle = 0, vjust = 0.5, margin = margin(r = 12)),
+      axis.line = element_line(size = 0.75),
       legend.position = "top"
     ) +
     scale_y_continuous(
       labels = scales::number_format(accuracy = 1, big = ",", prefix = "£")
     ) +
     xlab("Academic year end") +
-    ylab("Average revenue balance") +
+    ylab("Average\nrevenue\nbalance") +
```
[the remark this answers is not in the document]
Comment (rmbielby) [reply R/plotting.R:25]: Yep! Good call!

**Comment by rmbielby** [comment]
> Can we add the values of the line to the hover-over box on this one? image

Turned out it was actually a bit quirky to do this in ggiraph, so very useful to have checked out. Have added it now. You need to be quite precise with the hover to get the tooltip though, doesn't seem as zonally fluid with the tooltips as plotly.

**Review by cjrace — APPROVED** [review]
All looks good and should be nicer than plotly for performance and accessibility too, happy for you to merge!

One question that doesn't need addressing before merging - is it possible to give the tooltips a more forgiving hover area? Currently you have be very precisely on the data point, but it'd be nice if the area your cursor could be in to trigger the tooltip was slightly larger?

**Pull request merged**